katharsis-core/src/main/java/io/katharsis/core/internal/resource/AnnotationResourceInformationBuilder.java

```diff
@@ -137,7 +138,7 @@ public static String getResourceType(Type genericType, ResourceInformationBuilde
 			ResourceFieldType resourceFieldType = AnnotatedResourceField.getResourceFieldType(annotations);
 			String oppositeResourceType = resourceFieldType == ResourceFieldType.RELATIONSHIP ? getResourceType(getter.getGenericReturnType(), context) : null;
 			AnnotatedResourceField resourceField = new AnnotatedResourceField(jsonName, underlyingName, getter.getReturnType(), getter.getGenericReturnType(), oppositeResourceType, annotations);
-			if (Modifier.isStatic(getter.getModifiers())) {
+			if (Modifier.isStatic(getter.getModifiers()) || resourceField.isAnnotationPresent(JsonIgnore.class)) {
 				fieldWrappers.add(new ResourceFieldWrapper(resourceField, true));
```

Comment:
this if/else statement could be entirely replaced by simply substituting the predicated for the parameter that's different between the branches. 

I don't think you wrote all the surrounding code, but reducing complexity when/where you can should always be in your mind.